go-controller/pkg/ovn/obj_retry.go

```diff
@@ -416,23 +417,76 @@ func (oc *Controller) getResourceFromInformerCache(objType reflect.Type, key str
 	return obj, err
 }
 
-// Given an object and its type, recordAddEvent records the add event on this object. Only used for pods now.
+// Given an object and its type, recordAddEvent records the add event on this object.
 func (oc *Controller) recordAddEvent(objType reflect.Type, obj interface{}) {
 	switch objType {
 	case factory.PodType:
 		klog.V(5).Infof("Recording add event on pod")
 		pod := obj.(*kapi.Pod)
 		oc.podRecorder.AddPod(pod.UID)
+		metrics.GetConfigDurationRecorder().Start("pod", pod.Namespace, pod.Name)
+	case factory.PolicyType:
+		klog.V(5).Infof("Recording add event on network policy")
+		np := obj.(*knet.NetworkPolicy)
+		metrics.GetConfigDurationRecorder().Start("networkpolicy", np.Namespace, np.Name)
+	case factory.ServiceType:
```

Comment:
So we didn't apply the retry logic to services, because their controller already implements level-driven logic.

Is there a particular reason why you included this case block here and in the following functions (`recordUpdateEvent`, `recordDeleteEvent`, `recordSuccessEvent`)? This case block will never be executed because `WatchResource` is not applied to services. When you see the keyword services here it's actually `factory.PeerServiceType`, which corresponds to the services impacted by a network policy and we watch them with `WatchResource`.

Reply:
Mistake. I will remove.